In [28]:
import pandas as pd
import os
import numpy as np

#### Question: Is the new X8 bus route improving the commute?

##### *Null Hypothesis:* The average commute time of the new route is greater than or equal to the average commute time of the old route.

##### *Alternative Hypothesis:* The average commute time of the new route is less than the average commute time of the old route.

##### *Confidence Level:* I will use a significance level $\\alpha=0.05$

In [16]:
# mean and standard deviation for the existing commute
old_mu, old_sigma = 36, 6

In [10]:
# read in the data for the new bus route
url = 'https://raw.githubusercontent.com/fedhere/PUI2018_fb55/master/Lab4_fb55/times.txt'
new_route_data = pd.read_csv(url, delimiter = '\t', header=None)

In [14]:
new_route_data.columns = ['Commute_Time']

In [15]:
new_route_data.head()

Commute_Time
0     31.622239
1     32.821376
2     30.229101
3     31.413766
4     39.010550

In [26]:
#calculate the mean and the standard deviation of the new commute
new_mu = new_route_data.mean()
N = len(new_route_data)
print("The average new commute time is %f minutes and the N is %i"%(new_mu,N))

The average new commute time is 34.466162 minutes and the N is 100


#### Calculate the Z score

#### $Z = \frac{\frac{\mu_{pop} - \mu_{sample}}{\sigma}}{\sqrt{N}}$

In [29]:
Z = (old_mu - new_mu) / (old_sigma / np.sqrt(N))

In [31]:
print("The Z score is %f"%Z)

The Z score is 2.556397


#### The Z score is 2.6 which is greater than two standard deviations from the mean, which means that there is a less than 5% chance of randomly observing the new population mean, so we can reject the null hypothesis.